src/Confluent.Kafka/Consumer.cs

```diff
@@ -734,7 +732,11 @@ public void Stop()
         {
             if (consumerCts == null)
             {
-                throw new Exception("Consumer background poll loop not started - cannot stop.");
+                // Consumer background poll loop not started - cannot stop.
+                // While this might be considered an error there is no point
+                // in throwing an exception here since there are no side-effects
+                // of not being able to stop a non-Started instance.
+                return;
```

Comment:
on the other hand it indicates a logic error and I would argue that throwing an exception to point that out is a good thing. 

note: this has all been refactored in an open PR + I think there is a good chance we'll even cut background thread mode from the upcoming release as I think there might be a better factoring than what i've got (maybe better to take it outside the `Consumer` class).